src/main.rs

```diff
@@ -177,7 +190,10 @@ fn main() -> Result<()> {
         }
     }
 
-    Device::single()?.write(payload)?;
+    match args.transport {
+        TransportProtocol::Usb => UsbDevice::single()?.write(payload),
+        TransportProtocol::Ble => BleDevice::single().await?.write(payload).await,
```

Comment:
Let's keep the `main` function blocking and start an async runtime only when needed:
```suggestion
        TransportProtocol::Ble => tokio::runtime::Builder::new_current_thread()
            .enable_all()
            .build()?
            .block_on(async {
                BleDevice::single().await?.write(payload).await
            }),
```

Note: The `macros` feature of `tokio` is then no longer needed.